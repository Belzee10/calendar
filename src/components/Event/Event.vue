<template>
  <div
    class="event text-truncate text-capitalize ma-1 px-1"
    :class="getColorClass"
    @click="emitClick"
  >
    <v-icon v-if="isPriority" x-small>
      mdi mdi-star
    </v-icon>
    {{ name }}
  </div>
</template>

<script>
export default {
  name: 'Event',
  props: {
    id: {
      type: String,
      default: ''
    },
    name: {
      type: String,
      default: ''
    },
    // TODO not sure
    date: {
      type: Object,
      default: () => ({})
    },
    type: {
      type: String,
      default: ''
    },
    isPriority: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    getColorClass() {
      return this.type === 'work' ? 'grey' : 'blue';
    }
  },
  methods: {
    emitClick() {
      this.$emit('click-on-event', {
        id: this.id
      });
    }
  }
};
</script>

<style lang="scss" scoped>
.event {
  font-size: 12px;
}
</style>
