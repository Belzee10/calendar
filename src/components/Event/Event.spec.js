import { mount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import Event from './Event.vue';

const localVue = createLocalVue();

describe('Event.vue', () => {
  let vuetify;
  beforeEach(() => {
    vuetify = new Vuetify();
  });
  test('should emit "click" event with the correct payload', () => {
    const props = {
      id: '1'
    };
    const wrapper = mount(Event, {
      vuetify,
      localVue,
      propsData: props
    });
    wrapper.find('.event').trigger('click');
    expect(wrapper.emitted('click-on-event')[0]).toEqual([props]);
  });

  test('should set "blue" or "grey" colors depending of the type', () => {
    const props = {
      type: 'home'
    };
    const wrapper = mount(Event, {
      vuetify,
      localVue,
      propsData: props
    });
    expect(wrapper.find('.event').classes()).toContain('blue');
    expect(wrapper.find('.event').classes()).not.toContain('grey');
    wrapper.setProps({
      type: 'work'
    });
    expect(wrapper.find('.event').classes()).toContain('grey');
    expect(wrapper.find('.event').classes()).not.toContain('blue');
  });

  test('should render an star icon if the event is priority', () => {
    const props = {
      isPriority: true
    };
    const wrapper = mount(Event, {
      vuetify,
      localVue,
      propsData: props
    });
    expect(wrapper.find('.event').html()).toContain('mdi mdi-star');
  });
});
