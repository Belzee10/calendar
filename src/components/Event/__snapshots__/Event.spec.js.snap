// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event.vue should render correctly 1`] = `
<div
  class="event text-truncate text-capitalize ma-1 px-1 grey"
>
  <i
    aria-hidden="true"
    class="v-icon notranslate mdi mdi mdi mdi-star theme--light"
    style="font-size: 12px;"
  />
  
  

</div>
`;
